CRM/Sepa/Logic/Queue/Update.php

```diff
@@ -116,32 +121,36 @@ protected function __construct($cmd, $mode, $creditor_id = NULL, $offset = NULL,
       }
   }
 
-  public function run($context) {
+  public function run($context): bool {
+    $lock = SepaBatchLockManager::getInstance()->getLock();
+    if (!$lock->acquire(10, $this->asyncLockId)) {
+      throw new \RuntimeException('Unable to acquire lock');
+    }
+
     switch ($this->cmd) {
       case 'PREPARE':
         // nothing to do
         break;
 
       case 'CLOSE':
         CRM_Sepa_Logic_Batching::closeEnded();
-        CRM_Sepa_Logic_Settings::renewAsyncLock('sdd_async_update_lock', SDD_UPDATE_RUNNER_BATCH_LOCK_TIMOUT);
         break;
 
       case 'UPDATE':
-        if ($this->mode == 'OOFF') {
-          CRM_Sepa_Logic_Batching::updateOOFF($this->creditor_id, 'now', $this->offset, $this->limit);
+        if ($this->mode === 'OOFF') {
+          CRM_Sepa_Logic_Batching::updateOOFF($this->creditorId, 'now', $this->offset, $this->limit);
         } else {
-          CRM_Sepa_Logic_Batching::updateRCUR($this->creditor_id, $this->mode, 'now', $this->offset, $this->limit);
+          CRM_Sepa_Logic_Batching::updateRCUR($this->creditorId, $this->mode, 'now', $this->offset, $this->limit);
         }
```

Comment:
Let's apply code formatting while we're at it (`else` on new line, comparison value on left side).

Reply:
Changed.